org.adempiere.base/src/org/compiere/print/ReportEngine.java

```diff
@@ -1838,10 +1839,12 @@ static public ReportEngine get (Properties ctx, ProcessInfo pi, int windowNo)
 		int AD_PrintFormat_ID = 0;
 		boolean IsForm = false;
 		int Client_ID = -1;
+		
+		MPInstance instance = null;
 
 		//	Get AD_Table_ID and TableName
 		StringBuilder sql = new StringBuilder("SELECT rv.AD_ReportView_ID,rv.WhereClause,")
-			.append(" t.AD_Table_ID,t.TableName, pf.AD_PrintFormat_ID, pf.IsForm, pf.AD_Client_ID ")
+			.append(" t.AD_Table_ID,t.TableName, pf.AD_PrintFormat_ID, pf.IsForm, pf.AD_Client_ID, pi.AD_PInstance_ID ")
```

Comment:
This change is not needed, you already have the value from pi.getAD_Pinstance_ID().